tests/test_self.py

```diff
@@ -665,7 +665,7 @@ def test_fail_under(self):
                 "--enable=all",
                 join(HERE, "regrtest_data", "fail_under_plus6.py"),
             ],
-            code=0,
+            code=16,
```

Comment:
Why the change here ?

Reply:
Good point, I think I just saw that this was exactly testing `score == config.fail_under` and expected that I would now have to change the return value. Reverting now, but why wasn't this failing before?

Reply:
I'm not super familiar with this part of the code. Would you mind investigating a little ? Maybe you'll kill two bugs with one merge-request 😄

Reply:
This took a bit longer than expected but I think I fixed & expanded all regression tests in 376b3d0

The returned tests scores were different from the comment and I had to modify the regression test files a bit to get a stable score to test against. I covered all over, exact and under cases for positive and negative numbers

Reply:
Thanks a lot :)